torch/csrc/autograd/VariableTypeManual.cpp

```diff
@@ -305,7 +321,7 @@ Tensor VariableType::detach(const Tensor & self) const {
 
   }
   // <NON_GENERATED_CODE>
-  auto result = as_variable_ref(const_cast<Tensor&>(self)).detach();
+  auto result = as_variable_ref(const_cast<Tensor&>(self)).detach(); // NOLINT(cppcoreguidelines-pro-type-const-cast)
```

Comment:
This is sort of related to the changes above. What's going on here? Why do you need a non-const ref?